### PR TITLE
Fix Notice Dismiss

### PR DESCRIPTION
**Pull request by cmunozgar**
## Description
If you set `disableAnimation` to the `Notice` component, you cannot dismiss the notice because it relies on the `AnimationWrapper` to hide the Notice. This PR removes the animation properties before sharing them to the `AnimationWrapper`.

## Motivation and Context
We are using the Notice component to create a banner in Mastodon Profile pages to encourage customers to follow Buffer.

CleanShot 2023-02-01 at 16 19 21

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style and guidelines of this project. 
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the **CHANGELOG** document.
- [ ] I have added tests to cover my changes.
- [ ] I have performed a self-review of my own code
- [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] All new and existing tests passed.
- [ ] I have performed the accessibility audit of my UI changes according to the accessibility doc. 
- [ ] [Buffer Engineers] Someone from the Design team reviewed and approved my changes
- [ ] [Buffer Engineers] I have notified the BDS team of my changes in the #proj-design-system Slack channel

### CODE REVIEW

File: src/components/Notice/Notice.jsx [hunk 1]
```diff
@@ -103,6 +103,14 @@ function Notice({ children, dismiss, type, className, disableAnimation }) {
     onDismiss: dismiss,
   });
 
+  // We always need to wrap the Notice with AnimationWrapper because the dismiss function
+  // shows the Notice based on the dismissed property. If the animation is disabled the properties
+  // that control the animation are removed from animationProps.
+  if (disableAnimation) {
+    delete animationProps.stageInAnimation;
+    delete animationProps.stageOutAnimation;
+  }
```
Comment (guldenmw) [inline src/components/Notice/Notice.jsx:112]:
```suggestion
  const animationWrapperProps = disableAnimation ? {} : animationProps;
```

I'm not sure if this is actually better, but it feels like deleting the individual props specifically could lead to us missing one if someone makes changes to the `useAnimation` hook.

**Review by guldenmw — COMMENTED** [review]
Just left the one comment. Everything else seems okay.
Let me know what you think.

File: src/components/Notice/Notice.jsx (hunk 1 above)
Comment (guldenmw) [reply src/components/Notice/Notice.jsx:112]: The naming and alternative implementation don't really matter to me as much as deleting the props individually.

Comment (cmunozgar) [reply src/components/Notice/Notice.jsx:112]: I like the suggestion 👌🏼

**Review by guldenmw — APPROVED** [review]

**Pull request merged**